<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>960</width>
    <height>611</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Graph builder</string>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QFrame" name="PropertiesFrame">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>261</width>
      <height>611</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>10</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>10</number>
     </property>
     <item>
      <widget class="QScrollArea" name="scrollArea">
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>257</width>
          <height>549</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>2</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QToolBox" name="toolBox">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="page">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>257</width>
              <height>355</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Tabulation points</string>
            </attribute>
            <layout class="QFormLayout" name="formLayout_2">
             <property name="sizeConstraint">
              <enum>QLayout::SetDefaultConstraint</enum>
             </property>
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
             </property>
             <property name="horizontalSpacing">
              <number>6</number>
             </property>
             <property name="leftMargin">
              <number>7</number>
             </property>
             <property name="rightMargin">
              <number>7</number>
             </property>
             <item row="3" column="0">
              <widget class="QLabel" name="StartPointLabel">
               <property name="text">
                <string>Start:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QDoubleSpinBox" name="StartPointDoubleSpinBox">
               <property name="minimum">
                <double>-1000000000000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000000000000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="EndPointLabel">
               <property name="text">
                <string>End:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QDoubleSpinBox" name="EndPointDoubleSpinBox">
               <property name="minimum">
                <double>-1000000000000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000000000000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QLabel" name="PointsCountLabel">
               <property name="text">
                <string>Count:</string>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QSpinBox" name="PointsCountSpinBox">
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="maximum">
                <number>100000</number>
               </property>
              </widget>
             </item>
             <item row="6" column="0" colspan="2">
              <widget class="QCheckBox" name="OverrideYBoundsCheckBox">
               <property name="text">
                <string>Override Y bounds</string>
               </property>
              </widget>
             </item>
             <item row="9" column="0">
              <widget class="QLabel" name="YStartPointLabel">
               <property name="text">
                <string>Start:</string>
               </property>
              </widget>
             </item>
             <item row="10" column="0">
              <widget class="QLabel" name="YEndPointLabel">
               <property name="text">
                <string>End:</string>
               </property>
              </widget>
             </item>
             <item row="9" column="1">
              <widget class="QDoubleSpinBox" name="YStartPointDoubleSpinBox">
               <property name="minimum">
                <double>-1000000000000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000000000000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="10" column="1">
              <widget class="QDoubleSpinBox" name="YEndPointDoubleSpinBox">
               <property name="minimum">
                <double>-1000000000000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000000000000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="2">
              <widget class="QRadioButton" name="UseFunctionRadioButton">
               <property name="text">
                <string>&amp;Use</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
               <property name="autoExclusive">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QComboBox" name="FunctionChooserComboBox">
               <item>
                <property name="text">
                 <string>Choose current function</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>f(x) = Ax^2 + Bx + C</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>f(x) = Asin(Bx) + C</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>f(x) = Aln(Bx) + C</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>f(x) = 1 / ((Ax)^B) + C</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="page_2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>257</width>
              <height>355</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Paramteres</string>
            </attribute>
            <layout class="QFormLayout" name="formLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>A:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="AParamDoubleSpinBox">
               <property name="minimum">
                <double>-1000000000000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000000000000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>B:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QDoubleSpinBox" name="BParamDoubleSpinBox">
               <property name="minimum">
                <double>-1000000000000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000000000000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>C:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QDoubleSpinBox" name="CParamDoubleSpinBox">
               <property name="minimum">
                <double>-1000000000000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000000000000.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="page_3">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>257</width>
              <height>355</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Grid properties</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QComboBox" name="GridColorComboBox">
               <item>
                <property name="text">
                 <string>Choose color:</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Red</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Black</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>White</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="GridStyleComboBox">
               <item>
                <property name="text">
                 <string>Choose style</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Solid lines</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Dotted lines</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Dashed lines</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="page_4">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>257</width>
              <height>355</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Function line properties</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QComboBox" name="FunctionColorComboBox">
               <item>
                <property name="text">
                 <string>Choose color:</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Red</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Black</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>White</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="FunctionStyleComboBox">
               <item>
                <property name="text">
                 <string>Choose style</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Solid lines</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Dotted lines</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Dashed lines</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="page_5">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>257</width>
              <height>355</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Fourier Series</string>
            </attribute>
            <layout class="QFormLayout" name="formLayout_3">
             <property name="horizontalSpacing">
              <number>0</number>
             </property>
             <property name="verticalSpacing">
              <number>3</number>
             </property>
             <property name="leftMargin">
              <number>2</number>
             </property>
             <property name="topMargin">
              <number>15</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <item row="0" column="0" colspan="2">
              <widget class="QRadioButton" name="UseFouriersRadioButton">
               <property name="text">
                <string>&amp;Use</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0" colspan="2">
              <widget class="QPushButton" name="ImportButton">
               <property name="text">
                <string>Import</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QCheckBox" name="checkBox">
               <property name="text">
                <string>Keep loaded</string>
               </property>
              </widget>
             </item>
             <item row="5" column="0" colspan="2">
              <widget class="QPushButton" name="ExportButton">
               <property name="text">
                <string>Export</string>
               </property>
              </widget>
             </item>
             <item row="8" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Number of harmonics:</string>
               </property>
              </widget>
             </item>
             <item row="8" column="1">
              <widget class="QSpinBox" name="HarmonicsNumberSpinBox">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
              </widget>
             </item>
             <item row="9" column="0">
              <widget class="QCheckBox" name="ShowHarmonicsCheckBox">
               <property name="text">
                <string>Show harmonics</string>
               </property>
              </widget>
             </item>
             <item row="10" column="0">
              <widget class="QCheckBox" name="AnimateCheckBox">
               <property name="text">
                <string>Animate</string>
               </property>
              </widget>
             </item>
             <item row="11" column="0">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Animation speed:</string>
               </property>
              </widget>
             </item>
             <item row="12" column="0" colspan="2">
              <widget class="QSlider" name="AnimSpeedHorizontalSlider">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>200</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="DrawButton">
       <property name="text">
        <string>Draw</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
